google/cloud/spanner/value.cc

```diff
@@ -88,6 +88,73 @@ bool Equal(google::spanner::v1::Type const& pt1,
   }
 }
 
+std::ostream& StreamHelper(std::ostream& os, google::protobuf::Value const& v,
+                           google::spanner::v1::Type const& t) {
+  if (v.kind_case() == google::protobuf::Value::kNullValue &&
+      t.code() != google::spanner::v1::ARRAY &&
```

Comment:
I believe there's a difference between an empty array and a null array, and we probably want to print them differently too. I think all NULL values should print the string "NULL", including arrays and structs. And empty values should print as "[]" or "()". To make this work, I think you may just need to remove the two ARRAY and STRUCT conditions here.

WDYT?